mp2i/cogs/suggestions.py

```diff
@@ -1,218 +1,443 @@
+            title="**Fonctionnement des suggestions**",
             description=content,
             colour=0xFF66FF,
-            timestamp=datetime.now(),
         )
-        embed.set_thumbnail(url=guild.icon.url)
-        embed.set_footer(text=f"Généré par {self.bot.user.name}")
-        await ctx.send(embed=embed)
+        if guild.icon:
+            embed.set_thumbnail(url=guild.icon.url)
+
+        button = discord.ui.Button(
+            custom_id=f"suggestion:proposal",
+            style=discord.ButtonStyle.green,
+            label="Soumettre une suggestion"
+        )
+        view = discord.ui.View()
+        view.add_item(button)
 
-    @Cog.listener("on_message")
-    async def make_suggestion(self, msg) -> None:
+        message = await channel.send(embed=embed, view=view)
+        guild.suggestion_message_id = message.id
+
+    async def __make_suggestion(self, title, content, guild, user) -> None:
         """
-        Add reactions to a suggestion message and create a thread.
+        Create suggestion message, add reactions and then create a thread.
+
+        Parameters
+        ----------
+        title : str
+            Suggestion's title
+        content : str
+            Suggestion's content
+        user : discord.User
+            User who made the suggestion
         """
-        if msg.author.bot or isinstance(msg.channel, discord.DMChannel):
+        guild = GuildWrapper(guild)
+        if not guild:
+            logger.warning("Guild %s not found", guild.id)
             return
-        if msg.channel != GuildWrapper(msg.channel.guild).suggestion_channel:
+        if not guild.suggestion_channel:
+            logger.warning("Guild %s has no suggestion channel", guild.id)
             return
+        embed = discord.Embed(
+            title=title,
+            description=content,
+            colour=0x4286f4
+        )
+        if user.avatar:
+            embed.set_author(name=user.name, icon_url=user.avatar.url)
+        else:
+            embed.set_author(name=user.name)
+        embed.set_footer(text=f"\uD83D\uDD51 Suggestion non-traitée")
+        embed.timestamp = datetime.now()
+        msg = await guild.suggestion_channel.send(embed=embed)
         try:
             await msg.add_reaction("✅")
             await msg.add_reaction("❌")
-            await msg.channel.create_thread(
-                name=f"Suggestion de {msg.author.name}", message=msg
+            thread = await guild.suggestion_channel.create_thread(
+                name=title, message=msg
             )
+            await thread.add_user(user)
         except discord.errors.NotFound:
             pass
         database.execute(
             insert(SuggestionModel).values(
-                author_id=msg.author.id,
+                author_id=user.id,
                 date=datetime.now(),
                 guild_id=msg.guild.id,
+                title=title,
                 description=msg.content,
```

Comment:
```suggestion
                description=content,
```

Avant c'était un message donc ça fonctionnait, maintenant c'est un embed donc ça marche plus